testsuite/MDAnalysisTests/analysis/test_hole2.py

```diff
@@ -264,6 +264,34 @@ def test_deprecated_warning(self, hole, attrname):
             value = getattr(hole, attrname)
             assert value is hole.results[attrname]
 
+
+class TestOserror(object):
+
+    @pytest.fixture()
+    def universe(self):
+        return mda.Universe(MULTIPDB_HOLE)
+
+    def test_hole_method_oserror(self):
+        errmsg = exe_err.format(name='dummy_path', kw='executable')
+        with pytest.raises(OSError, match=errmsg):
+            hole = hole2.hole(PDB_HOLE, executable='dummy_path')
+
+    def test_hole_oserror(self, universe):
+        errmsg = exe_err.format(name='None', kw='executable')
+        with pytest.raises(OSError, match=errmsg):
+            h = hole2.HoleAnalysis(universe, executable='dummy_path')
```

Comment:
The actual message here doesn't seem to make much sense (``HOLE binary None not found. None must be on the PATH, or the path must provided with the keyword argument: executable``) -- I think this is yet another sneaky bug! (Sorry....)

Could you please change this so the message actually reads something like ``HOLE binary dummy_path not found. None must be on the PATH, or the path must provided with the keyword argument: executable`` ? Probably lines 792-795:

```
        hole = util.which(executable)
        if hole is None:
            raise OSError(errno.ENOENT, exe_err.format(name=hole,
                                                       kw='executable'))
```

Reply:
@lilyminium thanks for the review, I've done all the suggested changes